src/main/java/edu/ucsb/cs156/example/controllers/RecommendationController.java

```diff
@@ -0,0 +1,120 @@
+package edu.ucsb.cs156.example.controllers;
+
+import java.time.LocalDateTime;
+
+import javax.validation.Valid;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestBody;
+
+
+import edu.ucsb.cs156.example.entities.Recommendation;
+import edu.ucsb.cs156.example.errors.EntityNotFoundException;
+import edu.ucsb.cs156.example.repositories.RecommendationRepository;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import lombok.extern.slf4j.Slf4j;
+
+@Api(description = "Recommendation")
+@RequestMapping("/api/recommendation")
```

Comment:
The mapping should be: "/api/Recommendation". You might also need to change some test cases to reflect this update.